File: src/main/java/VenkatSample.java
import java.util.concurrent.CompletableFuture;

public class VenkatSample {
    public static void main(String[] args) {
        System.out.println(" main "+Thread.currentThread());
        createCompleteableFuture().thenAccept((data)-> printIt(data));
    }


    public static CompletableFuture<Integer> createCompleteableFuture() {
        System.out.println(" createCompleteableFuture "+Thread.currentThread());
        return CompletableFuture.supplyAsync(()->computeAsync());

    }

    public static Integer computeAsync() {
        System.out.println(" compute "+Thread.currentThread());
        return 2;
    }

    public static void printIt(Integer data){
        System.out.println(" printIt "+Thread.currentThread());
    }
}

File: src/main/kotlin/completeablefuture/VenkatSample.kt
package completeablefuture

import java.util.concurrent.CompletableFuture

fun createCompleteableFuture(): CompletableFuture<Int> {
    println(" createCompleteableFuture ${Thread.currentThread()}")
    return CompletableFuture.supplyAsync { computeAsync() }
}

fun computeAsync(): Int {
    return 2
}


fun main(args: Array<String>) {
    print("main ${Thread.currentThread()}")
    createCompleteableFuture().thenAccept { data -> printIt(data) }

}

fun printIt(data: Int?) {
    println("printIt ${Thread.currentThread()}")
    print(data)
}
